File: c++_2_oops_and_dsa/E_Look_Back.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;
void solve(){
    ll n;
    cin>>n;
    vector<ll> arr;
    for(ll i=0;i<n;i++){
        ll x;
        cin>>x;
        arr.push_back(x);
    }
    // solution
    ll ans = 0;
    ll p = arr[0];
    ll c;
    for(ll i=0;i<n;i++){
        c = arr[i];
        if(c>=p){
            p = c;
            continue;
        }
        float nval = (log(float(p)) - log(float(c)))/log(2);
        ll x = nval;
        ll z = 0;
        if(x == nval) z = nval;
        else z = x+1;
        p = c*pow(2,z);
        ans += z;
    }
    cout<<ans<<endl;
}
int main(){
    ll t;
    cin>>t;
    // cout<<pow(t,1);
    while(t--){
        solve();
    }
}

File: c++_2_oops_and_dsa/E_Look_Back.py
import math

def solve():
    n = int(input())
    arr = list(map(int, input().split()))
    ans = 0
    p = arr[0]
    for i in range(1, n):
        c = arr[i]
        if c >= p:
            p = c
            continue
        p_c = p / c
        nval = math.log2(p_c)
        x = int(nval)
        z = 0
        if x == nval:
            z = x
        else:
            z = x + 1
        p = c * (2 ** z)
        ans += z
    print(ans)

t = int(input())
while t > 0:
    solve()
    t -= 1
